Create a guide to configure Azure OpenID with Okteto (#259)
okteto/docs: src/content/self-hosted/guides/azure-active-directory/index.mdx

Copy the `Directory (tenant) ID` value`. You'll need it to configure your Okteto Instance.

<p align="center"><Image src="/docs/self-hosted/guides/azure-active-directory/images/tenant-id.png" alt="Copy the Tenant ID" /></p>

On the left menu, click on "Certificates & secrets". Create a "New client secret" and copy the `Value`. You'll need it to configure your Okteto Instance.

<p align="center"><Image src="/docs/self-hosted/guides/azure-active-directory/images/secrets.png" alt="Copy the Secret ID and Value" /></p>

On the lest menu, click on "API permissions" and grant the following permissions:

<p align="center"><Image src="/docs/self-hosted/guides/azure-active-directory/images/permissions.png" alt="Grant permissions" /></p>

## Configure Okteto

Once you have the Okta application ready, you'll need to update the configuration of your Okteto Self-Hosted instance. 

First, update the auth section of your Helm configuration file (e.g. `values.yaml` ) with the values you obtained on the first step of this guide:

```yaml
auth:
  openid:
    enabled: true
    clientId: $CLIENT_ID
    clientSecret: $CLIENT_SECRET
    endpoints:
      issuer: https://sts.windows.net/$TENANT_ID/

>       authorization: https://login.microsoftonline.com/<<tenant-id>>/oauth2/v2.0/authorize

Reviewer: ```suggestion
      authorization: https://login.microsoftonline.com/$TENANT_ID/oauth2/v2.0/authorize
```

    mapping:
      externalIDKey: email
```

> You can also store the value of clientSecret [as a secret directly in Kubernetes](/docs/self-hosted/install/deployment/#cloud-provider-secret).

> Then, [upgrade your installation](/docs/self-hosted/install/upgrade/) to apply the changes. We recommend that you upgrade to the same version that you already have to minimize the changes and help with your troubleshooting. 

Reviewer: ```suggestion
Then, [upgrade your installation](/docs/self-hosted/install/upgrade/) to apply the changes. We recommend that you upgrade to the same version that you already have to minimize the changes and help you troubleshoot any issues. 
```

> Once the upgrade has been completed, go to your instance and click on the **Log In** button to access your instance using Azure Active Directory as the authentication provider.

Reviewer: ```suggestion
Once the upgrade has been completed, go to your instance and click on the **Log In** button to access your instance. If everything is configured correctly,  Okteto will now use your Azure Active Directory for user authentication.
```

<p align="center"><Image src="/docs/self-hosted/guides/azure-active-directory/images/with-azure.png" alt="Log in with Azure Active Directory" /></p>

### Customize the attributes that Okteto consumes

By default, Okteto will use the following attributes below when creating a user:

- `Name` as the display name of the user.
- `Nickname` as the username of the user.
- `Email` as the email.
- `Picture URL` for the user's profile picture.

If you are not using some of those values, or you want to customize which attributes get mapped, you can customize that by using the `mapping` field inside the `openid` key in your Helm configuration file.

```yaml
auth:
  openid:
    enabled: true

>     clientId: <<client-id>>

Reviewer: ```suggestion
    clientId: $CLIENT_ID
```

>     clientSecret: <<client-secret>>

Reviewer: ```suggestion
    clientSecret: $CLIENT_SECRET
```

>     endpoints:
>       issuer: https://sts.windows.net/<<tenant-id>>/
>       authorization: https://login.microsoftonline.com/<<tenant-id>>/oauth2/v2.0/authorize

Reviewer: ```suggestion
    endpoints:
      issuer: https://sts.windows.net/$TENANT_ID/
      authorization: https://login.microsoftonline.com/$TENANT_ID/oauth2/v2.0/authorize
```
